### DISCUSSION TITLE
Auto detect browsers or operating systems

### DISCUSSION THREAD

**Discussion by Tom01810** [add2cal/add-to-calendar-button#396] [Ideas / Requests]
Hi,

I think we should implement a feature that can detect what browser or operating system the user is using.
Auto pull up the default add to the calendar screen for that operating system.

There is no need to select the calendar through the list.

There is no need for pop-up after popup.,.

Better user experience.

What do you think?

**Comment by jekuer** [comment]
How would you know which calendar the user is using?
This information is not bound to the OS or browser.

**Reply by Tom01810** [reply]
Because most people are using their phones for calendar these days.

As far as I know all iphone users use their ios native calendar.

And all my android friends use their google calendar.

**Reply by jekuer** [reply]
This would not count as data point at all for such a widely used product.
Think of all the business people using Microsoft products. Or simply check the market share of Outlook!

**Comment by Tom01810** [comment]
Or you can add it as an optional feature, which can turn on or off.

**Reply by jekuer** [reply]
Having it as an optional setting could be fair. Like specifying different options per category (desktop vs. mobile iOS vs. mobile Android)

**Comment by AKwoKWH** [comment]
I support this feature, add a option for the developer to turn on the auto open calendar feature.
Probably like below:
android phone, window computer  => use google calendar
ios phone, ios macbook => use apple calendar

This are the behaviour for most users.
Those using apple product will use the apple calendar, while others will use the google calendar.
Of coz it is even better to have a fine-grant option for the developer to decide the mappings, as I believe the behaviour is location by location and also depend on the target users (like if this is a format business event, very likely users use outlook or apple calendar)

my observation is that many companies give an company iphone to employee if not the using the employee's own phone with a 2nd profile for the company.
So in this case for busi purpose: 
Android phone => Outlook (as most busi is using MS system)
iPhone => Apple Calendar

**Reply by jekuer** [reply]
It is on the roadmap, but also a very big change if we want to do it right. So, do not expect it any time soon.
However, you can still build something like this, by checking for the OS yourself and specifying options dynamically.

**Reply by AKwoKWH** [reply]
Yes, I am now putting it on my js code.
First determine the system os, then put the parameter in the `options`, either "Google" or "Apple"

An issue here I encountered is that I also want to check the browser as well and it seem very hard to do so on ios system.
For example I am using brave browser on iphone, there are no identifier and my code always return it is safari browser.

My plan is like this:
```
ios + safari => "Apple"
ios + chrome/brave => "Google"
ios + edge => "Outlook.com"
android => "Google"
```
